src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -73,34 +73,49 @@ function assembleArrow(
     borderColor = convertToColorString(borderColor);
     const arrowPos = mirrorPos(arrowPosition);
     const arrowSize = Math.max(Math.round(borderWidth) * 1.5, 6);
-    let positionStyle = '';
-    let transformStyle = CSS_TRANSFORM_VENDOR + ':';
+
+    const arrowClasses = ['tooltip-arrow'];
+    const transformClasses = [];
     let rotateDeg;
-    if (indexOf(['left', 'right'], arrowPos) > -1) {
-        positionStyle += 'top:50%';
-        transformStyle += `translateY(-50%) rotate(${rotateDeg = arrowPos === 'left' ? -225 : -45}deg)`;
+
+    if (['left', 'right'].includes(arrowPos)) {
+        arrowClasses.push('tooltip-arrow-horizontal');
+        transformClasses.push(`tooltip-arrow-rotate-${rotateDeg = arrowPos === 'left' ? -225 : -45}`);
     }
     else {
-        positionStyle += 'left:50%';
-        transformStyle += `translateX(-50%) rotate(${rotateDeg = arrowPos === 'top' ? 225 : 45}deg)`;
+        arrowClasses.push('tooltip-arrow-vertical');
+        transformClasses.push(`tooltip-arrow-rotate-${rotateDeg = arrowPos === 'top' ? 225 : 45}`);
+    }
+
+    function calculateArrowOffset(rotatedWH: number, borderWidth: number, arrowWH: number) {
+        return Math.round(
+            (
+                ((rotatedWH - Math.SQRT2 * borderWidth) / 2
+                    + Math.SQRT2 * borderWidth
+                    - (rotatedWH - arrowWH) / 2)
+                * 100
+            ) / 100
+        );
     }
+
+    function getColorClassName(color: ZRColor) {
+        const colorValue = convertToColorString(color);
+        return colorValue.replace(/[^a-zA-Z0-9]/g, '');
+    }
+
     const rotateRadian = rotateDeg * Math.PI / 180;
     const arrowWH = arrowSize + borderWidth;
     const rotatedWH = arrowWH * Math.abs(Math.cos(rotateRadian)) + arrowWH * Math.abs(Math.sin(rotateRadian));
-    const arrowOffset = Math.round(((rotatedWH - Math.SQRT2 * borderWidth) / 2
-        + Math.SQRT2 * borderWidth - (rotatedWH - arrowWH) / 2) * 100) / 100;
-    positionStyle += `;${arrowPos}:-${arrowOffset}px`;
-
-    const borderStyle = `${borderColor} solid ${borderWidth}px;`;
-    const styleCss = [
-        `position:absolute;width:${arrowSize}px;height:${arrowSize}px;z-index:-1;`,
-        `${positionStyle};${transformStyle};`,
-        `border-bottom:${borderStyle}`,
-        `border-right:${borderStyle}`,
-        `background-color:${backgroundColor};`
-    ];
-
-    return `<div style="${styleCss.join('')}"></div>`;
```

Comment:
Enabling a strict policy for the style-src directive in CSP offers several security benefits for your web application:
- XSS attacks aim to inject malicious scripts into your website. By restricting stylesheet sources, you prevent attackers from embedding malicious code within stylesheets and potentially compromising user data or website functionality.
- A strict policy grants you more control over the styles applied to your website. You determine the legitimate sources for stylesheets, preventing unauthorized styles from altering your website's appearance or behavior.

Reply:
styles properties that are set directly on the element's style property will not be blocked, allowing users to safely manipulate styles via JavaScript. e.g. this, document.body.style.background = 'green'; won't cause an unsafe-inline violation.

Reply:
Hello @Ovilia ,
Could you review this PR or help me find the right person?

Reply:
met the same issue , insert a styled item to dom will violate the unsafe-inline policy